src/functions/contactsMenu/index.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import logger from '../../utils/logger.js'
+
+// Taken from \OC\Contacts\ContactsMenu\Entry::jsonSerialize
+export interface ContactsMenuEntry {
+	id: number|string|null,
+	fullName: string,
+	avatar: string|null,
+	topAction: object|null,
+	actions: object[],
+	lastMessage: '',
+	emailAddresses: string[],
+	profileTitle: string|null,
+	profileUrl: string|null,
+	status: string|null,
+	statusMessage: null|string,
+	statusMessageTimestamp: null|number,
+	statusIcon: null|string,
+	isUser: boolean,
+	uid: null|string,
+}
+
+export interface ContactsMenuAction {
+	id: string,
+	displayName: (entry: ContactsMenuEntry) => string,
+	enabled: (entry: ContactsMenuEntry) => boolean,
+	iconSvg: (entry: ContactsMenuEntry) => string,
+	callback: (entry: ContactsMenuEntry) => void,
+}
+
+/**
+ * Register a contacts and avatar menu action that will invoke the given callback on click.
+ *
+ * @param {ContactsMenuAction} action The action to register
+ */
+export function registerContactsMenuAction(action: ContactsMenuAction): void {
+	window._nc_contacts_menu_hooks ??= {}
+
+	if (window._nc_contacts_menu_hooks[action.id]) {
+		logger.error(`ContactsMenu action for id ${action.id} has already been registered`, {
+			action,
+		})
+		return
+	}
+
+	window._nc_contacts_menu_hooks[action.id] = action
+}
```

Comment:
that makes it not available for other repos ?
Isn't there a library that would be fit to receive that API ?

Reply:
Hilarious, I thought I was in /server :eyes: